docs/src/dev/custom_metal_kernels.rst

```diff
@@ -76,6 +76,9 @@ Putting this all together, the generated function signature for ``myexp`` is as
 
   template [[host_name("custom_kernel_myexp_float")]] [[kernel]] decltype(custom_kernel_myexp_float<float>) custom_kernel_myexp_float<float>;
 
+Note: ``grid`` and ``threadgroup`` are parameters to the Metal `dispatchThreads <https://developer.apple.com/documentation/metal/mtlcomputecommandencoder/2866532-dispatchthreads>`_ function.
+This means we will launch ``mx.prod(grid)`` threads, subdivided into ``threadgroup`` size threadgroups.
```

Comment:
This could be a good spot to put a little warning here to make sure the thread group dimension is always less than or equal to the corresponding grid dimension.

Something like: 

> For optimal performance, each thread group dimension should be less than or equal to the corresponding grid dimension.